template/src/components/DebugMenu.tsx

```diff
@@ -0,0 +1,276 @@
+import BottomSheet, {BottomSheetFlatList} from '@gorhom/bottom-sheet'
+import React, {FC, ReactNode, useCallback, useMemo, useRef, useState} from 'react'
+import {
+  Modal,
+  Pressable,
+  SafeAreaView,
+  ScrollView,
+  View,
+  Text,
+  Button,
+  StyleSheet,
+  useWindowDimensions,
+  ListRenderItem,
+  ListRenderItemInfo,
+} from 'react-native'
+import {getApplicationName, getBuildNumber, getDeviceId} from 'react-native-device-info'
+import Draggable from './Draggable'
+
+import Config, {BOTTOM_SHEET_TYPE, CODEPUSH_KEYS, EXTRA_QA_ENVS} from '../constants/configs'
+
+import {InfoMenu, InfoMenuRow, InfoMenuLink} from './InfoMenu'
+
+import {colors, fonts, metrics} from '../themes'
+
+import {localize} from '../locale/I18nConfig'
+import {useDispatch, useSelector} from 'react-redux'
+import {getApiUrl, getCodePushKey} from '../store/selectors'
+import {appActions} from '../store/reducers'
+import {BottomSheetMethods} from '@gorhom/bottom-sheet/lib/typescript/types'
+
+interface Props {
+  title: string
+  children?: ReactNode
+}
+
+const debugMenuSize = 50
+
+const AppInfoSection: FC = () => {
+  const appName = useMemo(() => getApplicationName(), [])
+  const buildNumber = useMemo(() => getBuildNumber(), [])
+  const deviceId = useMemo(() => getDeviceId(), [])
+
+  const infos = useMemo(
+    () => [
+      {title: localize('debug.deviceId'), description: deviceId},
+      {title: localize('debug.appName'), description: appName},
+      {title: localize('debug.buildNumber'), description: buildNumber},
+      {title: localize('debug.appVersion'), description: Config.appVersion},
+      {title: localize('debug.bundleId'), description: Config.appBundleID},
+      {title: localize('debug.appEnv'), description: Config.APP_ENV || 'N/A'},
+    ],
+    [appName, buildNumber, deviceId],
+  )
+
+  return (
+    <View style={styles.section}>
+      <Text style={styles.h3}>{localize('debug.info')}</Text>
+      <View style={styles.content}>
+        {infos.map(({title, description}) => (
+          <InfoMenuRow style={styles.infoMenu} title={title} description={description} />
+        ))}
+      </View>
+    </View>
+  )
+}
+
+const EnvironmentSection: FC<Props> = ({title, children}) => (
+  <View style={styles.section}>
+    <Text style={styles.h3}>{title}</Text>
+    <View style={styles.content}>{children}</View>
+  </View>
+)
+
+const DebugMenu: FC = () => {
+  const [modalVisible, setModalVisible] = useState(false)
+  const openModal = useCallback(() => setModalVisible(true), [])
+  const closeModal = useCallback(() => setModalVisible(false), [])
+
+  const dispatch = useDispatch()
+
+  const codePushKey = useSelector(getCodePushKey)
+
+  const currentApiUrl = useSelector(getApiUrl)
+
+  const [bottomSheetType, setBottomSheetType] = useState<string>()
+
+  const dimensions = useWindowDimensions()
+
+  const bottomSheetRef = useRef<BottomSheetMethods>(null)
+
+  const snapPoints = useMemo(() => ['95%'], [])
```

Comment:
Move this to a constants outside of component, so we don't have to use `useMemo`